[stable/goldilocks] Add pod priority class for goldilocks

harshbhardwaj5 opened the pull request:
**Why This PR?**
_a short description of why this PR is needed_
There is no pod priority class support available for goldilocks start so this change add pod priority class in goldilocks deployment 
Fixes  https://github.com/FairwindsOps/charts/issues/1544

**Changes**
Changes proposed in this pull request:
* The addition of a pod priority class allows users to specify their own values based on their requirements, as there are no hardcoded defaults.


**Checklist:**

* [x] I have included the name of the chart in the title of this PR in square brackets i.e. `[stable/goldilocks]`.
* [x] I have updated the chart version in `Chart.yaml` following Semantic Versioning.
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] Any new values have been added to the README for the Chart, or `helm-docs --sort-values-order=file` has been run for the charts that support it.

harshbhardwaj5 commented:
How to get the `ci/circleci: check-helm-docs` erroe cleared out not able to understand where is the issue is ?

harshbhardwaj5 commented:
@sudermanjr  any help on this

harshbhardwaj5 commented:
@sudermanjr can you please approve

vitorvezani commented on stable/goldilocks/templates/controller-deployment.yaml:
```diff
@@ -43,6 +44,10 @@ spec:
       {{- end }}
+      {{- if .Values.controller.priorityClassName }}
+      priorityClassName: 
+        {{- toYaml .Values.controller.priorityClassName | nindent 8 }}
```
```suggestion
        {{- toYaml . | nindent 8 }}
```

vitorvezani commented:
Hi @harshbhardwaj5, added a couple of suggestions to keep consistent with other configuration

harshbhardwaj5 commented:
> Hi @harshbhardwaj5 -- thanks for the contribution, added a couple of suggestions to keep consistent with other configuration

Hi thanks for suggestions, fixed it 
Kindly approve @vitorvezani

harshbhardwaj5 commented:
I think @sudermanjr review is mandatory for it to merge

harshbhardwaj5 commented:
@sudermanjr  Hi can we get a approval here

harshbhardwaj5 commented:
Hi @sudermanjr  , Can we get this merged?
